dosbox-staging/dosbox-staging
# Kill the transliteration/romanization rules with fire 🔥🔥🔥

# Description

When introducing our Unicode engine, I have added several rules to transliterate Cyrillic/Greek/etc. strings to Latin-only ones.  This romanization scheme was very limited - it could only replace a non-Latin letter with a single Latin character, and the rules were the same for all the languages (while in reality rules for i..e. Russian and Bulgarian should be different) - but it served the purpose well, to keep the translated messages readable no matter what code page was set.

This rule set is now obsolete, as we fall back to English messages if the code page does not suite the non-Latin translation.

This PR removes almost all the transliteration rules (with very few exceptions, like the digits or homograph punctuation marks), as they are probably useless now.

I have kept the code point definitions in `ASCII.TXT` file on purpose:
- this is an indication that the code point is fully supported by the engine (we have decomposition rules, upper/lower case conversion rules, etc.)
- in the future I intend to extend this file with additional data, needed to generate DOS collation tables


# Release notes

(no need to advertise this - there is already a release notes proposal for using English as a fallback)


# Manual testing

1. Copy some Greek text to your host OS clipboard, like this piece of [Wikipedia](https://el.wikipedia.org/wiki/DOSBox):
> Το DOSBox είναι εξομοιωτής που εξομοιώνει έναν συμβατό υπολογιστή IBM PC εκτελώντας ένα λειτουργικό σύστημα DOS, μαζί με συμβατές κάρτες γραφικών και ήχου, που σημαίνει πως κανονικά προγράμματα DOS, συμπεριλαμβανόμενων και παιχνιδιών υπολογιστή, παρέχουν ένα περιβάλλον στο οποίο μπορούν να τρέξουν σωστά, παρόλο που οι σύγχρονοι υπολογιστές έχουν σταματήσει να υποστηρίζουν αυτό το περιβάλλον. Το DOSBox είναι ελεύθερο λογισμικό, γραμμένο κυρίως στη γλώσσα C++ και διανέμεται υπό την Γενική Άδεια Δημόσιας Χρήσης GNU.

2. Inside DOSBox, execute command `keyb us`, afterwards execute command `clip` - it should paste the text, replacing most Greek letters with question marks:
![Screenshot-US](https://github.com/user-attachments/assets/3607e73d-6b05-49c1-9b07-50a0a2291fef)

3. Inside DOSBox, execute command `keyb el`, afterwards execute command `clip` - it should paste the text, keeping the Greek letters:
![Screenshot-EL](https://github.com/user-attachments/assets/3d82f981-e144-4ff3-84c6-c937fee4b677)

The change has been manually tested on:

- [x] Linux (trivial)


# Checklist

I have:
